ulib/FStar.Array.fsti

```diff
@@ -0,0 +1,126 @@
+(*
+   Copyright 2008-2018 Microsoft Research
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*)
+
+module FStar.Array
+
+(**
+F* standard library mutable arrays module.
+
+@summary Mutable arrays
+*)
+
+open FStar.All
+open FStar.Seq
+open FStar.Ref
+
+#set-options "--max_fuel 0 --initial_fuel 0 --initial_ifuel 0 --max_ifuel 0"
+
+val array (a:Type0) : Type0
+
+val as_ref (#a:Type0) (arr:array a) : GTot (ref (seq a))
+
+let sel (#a:Type0) (h:heap) (s:array a) : GTot (seq a) = Heap.sel h (as_ref s)
+
+let contains (#a:Type0) (h:heap) (s:array a) : GTot bool =
+  FStar.StrongExcludedMiddle.strong_excluded_middle (Heap.contains h (as_ref s))
```

Comment:
Why bool and StrongExcludedMiddle instead of just a prop?

Use of this axiom in an interface is particularly troubling, since all clients will implicitly have the axiom in scope too.

Reply:
It (and the other instance of the axiom) was already like that but yeah, now it is in the client code. I will change it.